cpp/src/arrow/flight/flight_test.cc

```diff
@@ -1186,6 +1188,159 @@ class TestBasicHeaderAuthMiddleware : public ::testing::Test {
   std::shared_ptr<BearerAuthServerMiddlewareFactory> bearer_middleware_;
 };
 
+// This test has functions that allow adding and removing cookies from a list of cookies.
+// It also automatically adds cookies to an internal list of tracked cookies when they
+// are passed to the middleware.
+class TestCookieMiddleware : public ::testing::Test {
```

Comment:
This is a general comment, but there's a lot of logic here, and some of it duplicates the logic in the middleware class itself. Can we factor out the cookie parsing into an internal header and add direct tests of that logic, and perhaps reuse some of it here (e.g. splitting and parsing cookies) instead of duplicating it? That would also let us directly test the cookie parsing, instead of indirectly through these tests.